clients/src/main/java/org/apache/kafka/clients/producer/internals/ProducerBatch.java

```diff
@@ -146,44 +146,61 @@ private boolean tryAppendForSplit(long timestamp, ByteBuffer key, ByteBuffer val
 
     /**
      * Abort the batch and complete the future and callbacks.
-     *
      * @param exception The exception to use to complete the future and awaiting callbacks.
      */
-    public void abort(RuntimeException exception) {
+    public boolean abort(RuntimeException exception) {
```

Comment:
The method never returns false. We can keep it as void if so.

Reply:
@becketqin return type restored to `void`